Functions/homework _functions.py

```diff
@@ -0,0 +1,283 @@
+"""
+
+
+def get_pascal_triangle(n: int) -> list:
+    triangle = []
+
+    for i in range(n):
+        row = [1]
+        if i > 0:
+            for j in range(1, i):
+                row.append(triangle[i-1][j-1] + triangle[i-1][j])
+            row.append(1)
+        triangle.append(row)
+
+    return triangle
+
+print(get_pascal_triangle(6))
+
+"""
+5. You need to write a program to find whether a given string starts with a given character using Lambda.
+The function takes a word and a letter as input.
+
+Expected output:
+    True
+    False
+"""
+
+starts_with = lambda word,char: word[0]==char 
+print(starts_with('Python', 'P'))
+print(starts_with('Java', 'P'))
+
+"""
+6. You need to write a program to create Fibonacci series upto n using Lambda and reduce. It's a hard task.
+The function takes as input the number of Fibonacci numbers to be calculated.
+
+Expected output:
+    [0, 1, 1, 2, 3, 5]
+"""
+
+from functools import reduce
+
+#base condition [0, 1]
+fib_series =  lambda n: reduce(lambda acc, _: acc + [acc[-1] + acc[-2]], range(n - 2), [0, 1])
+'''n=6:
+init: 
+acc=[0,1]
+first iter: [0,1,1] (0+1)
+sec iter: [0,1,1,2] (1+1)
+third iter : [0,1,1,2, 3] (1+2)...
+'''
+
+print(fib_series(6))
+
+"""
+7. You need to write a program to find intersection of two given arrays using Lambda and filter.
+You are given two lists, the result will be a list-intersection of the given ones.
+
+Expected output:
+    [1, 2, 8, 9]
+"""
+
+first = [1, 2, 3, 5, 7, 8, 9, 10]
+second = [1, 2, 4, 8, 9]
+result = list(filter(lambda i: i in first, second))
+print(result)
+
+"""
+8. You need write a program to find palindromes in a given list of strings using Lambda and filter.
+According Wikipedia - A palindromic number or numeral palindrome is a number that remains the same
+when its digits are reversed. Like 16461, for example, it is "symmetrical".
+
+Expected output:
+    ['php', 'sagas', 'repaper', 'madam', 'level']
+"""
+
+words = ["php", "sagas", "Python", "abcdefg", "Java", "repaper", "madam", "level"]
+result = list(filter(lambda word: word[-1::-1]==word,words))
+#print(result)
+
+"""
+9.  You need write a Python program to add three given lists using Python map and lambda.
+You are given three lists, the result will be a list with numbers that are the sum of 
+the first values of the given lists, the second values, the third...
+
+Expected output:
+    [12, 15, 18]
+"""
+
+nums_1 = [1, 2, 3] #x
+nums_2 = [4, 5, 6] #y
+nums_3 = [7, 8, 9] #z
+result = list(map(lambda x,y,z: x+y+z, nums_1, nums_2,nums_3))
+print(result)
+
+"""
+10. You need a program to make a chain of function decorators (bold, italic, underline etc.).
+Apply the string formatting that is given as input to the function.
+
+Expected output:
+    <b><i><u>Hello world</u></i></b>
+"""
+
+
+def make_bold(func):
+    def wrapper():
+        result=func()
+        return ('<b>'+ result+ '</b>')
+    return wrapper
+
+
+
+def make_italic(func):
+    def wrapper():
+        result=func()
```

Comment:
also make sure you are using proper spacing in your code, spaces is a part of code style
`result=func()`  incorrect
`result = func()` correct